go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
+	ParameterWithConfig
+}
+
+// NewCheckpoint creates a new checkpoint.
+func NewCheckpoint(idx int, cpPath string, e *EtcdClient) (*Checkpoint, error) {
```

Comment:
As the offline discuss, I will add `Load` and `Save` method for the struct `Checkponit`:)